src/ManageClient.php

```diff
@@ -62,6 +62,26 @@ public function deactivate(string $id): void
         $this->sendRequest(new Request('POST', "manage/{$id}/deactivate", [], '{}'));
     }
 
+    /**
+     * @return iterable<Project>
+     */
+    public function listProjects(?ListProjectsOptions $options): iterable
+    {
+        $query = $options ? $options->toQueryParameters() : [];
+        $nextPageToken = '';
+
+        do {
+            $query['nextPageToken'] = $nextPageToken;
+            $response = $this->sendRequest(new Request('GET', '/manage/projects?'.http_build_query($query)));
+
+            foreach ($response['data'] ?? [] as $projectData) {
+                yield Project::fromArray($projectData);
+            }
+
+            $nextPageToken = $response['nextPageToken'] ?? '';
+        } while (!empty($nextPageToken));
```

Comment:
oooh, an old school do - while loop :)

Reply:
It's not that old school. I use this patter quite often when hiding API pagination using generators and I don't consider myself oldschool :wink: